pkg/destroy/baremetal/baremetal.go

```diff
@@ -36,8 +43,66 @@ func (o *ClusterUninstaller) Run() error {
 // New returns bare metal Uninstaller from ClusterMetadata.
 func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.Destroyer, error) {
 	return &ClusterUninstaller{
+		ClusterID:               metadata.ClusterID,
 		LibvirtURI:              metadata.ClusterPlatformMetadata.BareMetal.LibvirtURI,
 		BootstrapProvisioningIP: metadata.ClusterPlatformMetadata.BareMetal.BootstrapProvisioningIP,
 		Logger:                  logger,
 	}, nil
 }
+
+// deleteStoragePool destroys, deletes and undefines any storagePool left behind during the creation
+// of the bootstrap VM
+func (o *ClusterUninstaller) deleteStoragePool(conn *libvirt.Connect) error {
+	o.Logger.Debug("Deleting baremetal bootstrap volumes")
+
+	pools, err := conn.ListStoragePools()
+	if err != nil {
+		return errors.Wrap(err, "list storage pools")
+	}
+
+	for _, pname := range pools {
+		// delete only pool names that contains cluster name and 'bootstrap' string
+		if !strings.Contains(pname, "-bootstrap") && !strings.Contains(pname, o.ClusterID) {
+			continue
+		}
+
+		pool, err := conn.LookupStoragePoolByName(pname)
```

Comment:
Actually we know the storage pool by name, don't we? Can't we skip iterating over all the storage pools and just call

```go
conn.LookupStoragePoolByName(fmt.Sprintf("%s-bootstrap", o.ClusterID))
```

Reply:
Fixed in the latest update. Creating a pname based on InfraID, so it will just match one cluster.